qa/L0_python_api/test.sh

```diff
@@ -51,7 +51,14 @@ fi
 
 
 FRONTEND_TEST_LOG="./python_kserve.log"
-python -m pytest --junitxml=test_kserve.xml test_kserve.py > $FRONTEND_TEST_LOG 2>&1
+# TODO: [DLIS-7215] Run tritonclient.grpc as separate process
+# Currently, running tritonclient.grpc with tritonserver in the same process,
+# it will non-deterministically abort/crash without being able to be caught by pytest.
+# This is because fork() is called by tritonserver on model load,
+# which attempts to fork the imported libraries and their internal states,
+# and cygrpc (dependency of tritonclient.grpc) does not officially support fork().
```

Comment:
> However, **if the application only instantiate gRPC Python objects after calling fork(), then fork() will work normally**, since there is no C extension binding at this point.

If the claim is that we're calling `fork()` during model load, aren't we instantiating grpc (client) after the `fork()` if server/service have already started up?

Can you ever reproduce this by only running a single test case repeatedly?

Reply:
From, my understanding it is an internal state set upon import:

```python

import tritonclient.grpc as grpcclient

while True:
     server = utils.startup_server()
     sleep(1)
     utils.teardown_server(server)
```